test/box-luatest/gh_8650_add_is_sync_option_test.lua

```diff
@@ -0,0 +1,319 @@
+local t = require('luatest')
+local cluster = require('luatest.replica_set')
+local server = require('luatest.server')
+local net = require('net.box')
+local msgpack = require('msgpack')
+
+local g = t.group('gh_8650')
+
+g.before_each(function(cg)
+    cg.cluster = cluster:new({})
+    local box_cfg = {
+        memtx_use_mvcc_engine = true,
+        replication = {
+            server.build_listen_uri('master', cg.cluster.id),
+            server.build_listen_uri('replica', cg.cluster.id)
+        },
+        replication_synchro_timeout = 0.2,
```

Comment:
I suggest you leave the default value for `replication_synchro_timeout` here.

When you're testing a rollback you may re-set `replicaiton_synchro_timeout` to a tiny value like `0.01`.

Reply:
Let's also set `replication_synchro_timeout` to a tiny value before testing rollbacks on master. So that the test doesn't run for 30 seconds.

Like this:
```diff
diff --git a/test/box-luatest/gh_8650_add_is_sync_option_test.lua b/test/box-luatest/gh_8650_add_is_>
index 2693a2bdf5..969a63b304 100644
--- a/test/box-luatest/gh_8650_add_is_sync_option_test.lua
+++ b/test/box-luatest/gh_8650_add_is_sync_option_test.lua
@@ -107,9 +107,12 @@ g.test_box_begin_commit_is_sync = function(cg)
     err_msg = "Quorum collection for a synchronous transaction is timed out"
     t.assert_error_msg_content_equals(err_msg, function()
         cg.master:exec(function()
+            local timeout = box.cfg.replication_synchro_timeout
+            box.cfg{replication_synchro_timeout = 0.01}
             box.begin({is_sync = true})
                 box.space.test:insert{10, 'is_sync = true'}
             box.commit()
+            box.cfg{replication_synchro_timeout=timeout}
         end)
     end)
 
```